File: java/logic-1/in1To10.java
/* Given a number n, return true if n is in the range 1..10, inclusive. 
 * Unless "outsideMode" is true, in which case return true if the number is 
 * less or equal to 1, or greater or equal to 10.
 */
public boolean in1To10(int n, boolean outsideMode) {
    if(outsideMode)
        return n <= 1 || 10 <= n;
          
    return 1 <= n && n <= 10;
}

File: python/logic-1/in1to10.py
# Given a number n, return True if n is in the range 1..10, inclusive. Unless 
# "outsideMode" is True, in which case return True if the number is less or 
# equal to 1, or greater or equal to 10.
def in1to10(n, outside_mode):
    if outside_mode:
        return n <= 1 or n >= 10
          
    return 1 <= n and n <= 10
